datasetinsights/io/downloader/http_downloader.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import os
+import zipfile
+
+from datasetinsights.io.download import (
+    download_file,
+    get_checksum_from_file,
+    validate_checksum,
+)
+from datasetinsights.io.downloader.base import DatasetDownloader
+from datasetinsights.io.exceptions import ChecksumError
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPDownloader(DatasetDownloader, protocol="http://"):
+    """ This class is used to download data from any HTTP or HTTPS public url
+        and perform function such as downloading the dataset and checksum
+        validation.
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def download(self, source_uri, output, checksum_file=None, **kwargs):
+        """ This method is used to download the dataset from HTTP or HTTPS url.
+
+        Args:
+            source_uri (str): This is the downloader-uri that indicates where
+                              the dataset should be downloaded from.
+
+            output (str): This is the path to the directory where the download
+                          will store the dataset.
+
+            checksum_file (str): This is path of the txt file that contains
```

Comment:
Are we going to provide checksum files of the datasets in the public GCS bucket?

Reply:
Yes.